spec/lib/legacy_activity_spec.rb

```diff
@@ -0,0 +1,156 @@
+require "rails_helper"
+require "legacy_activity"
+
+RSpec.describe LegacyActivity do
+  let(:delivery_partner) { create(:delivery_partner_organisation) }
+  let(:activity_node_set) do
+    Nokogiri::XML(
+      File.read("#{Rails.root}/spec/fixtures/activities/uksa/single_activity.xml"),
+      nil,
+      "UTF-8"
+    ).xpath("//iati-activity").first
+  end
+
+  describe "#elements" do
+    it "returns the nokogiri nodes for this document" do
+      result = described_class.new(activity_node_set: activity_node_set, delivery_partner: delivery_partner).elements
+      expect(result.class).to eql(Nokogiri::XML::NodeSet)
+    end
+  end
+
+  describe "#to_xml" do
+    it "returns the xml representation for this document" do
+      legacy_xml = File.read("#{Rails.root}/spec/fixtures/activities/uksa/individual_activity.xml")
+
+      result = described_class.new(activity_node_set: activity_node_set, delivery_partner: delivery_partner).to_xml
+
+      expect(result.squish).to eql(legacy_xml.squish)
+    end
+  end
+
+  describe "#identifier" do
+    context "when there is an identifier element" do
+      it "returns the full IATI identifier" do
+        activity_xml = <<~XML
+          <?xml version="1.0" encoding="UTF-8"?>
+          <iati-activities generated-datetime="2019-09-25T22:01:44.470000+00:00" version="2.03">
+            <!--Data generated by IATI CoVE. Built by Open Data Services Co-operative: http://iati.cove.opendataservices.coop/-->
+            <iati-activity default-currency="GBP" hierarchy="2" last-updated-datetime="2019-09-25T22:01:44.470000+00:00">
+              <iati-identifier>GB-GOV-13-GCRF-UKSA_NS_UKSA-019</iati-identifier>
+            </iati-activity>
+          </iati-activities>
+        XML
+
+        activity_node_set = Nokogiri::XML(activity_xml, nil, "UTF-8").xpath("//iati-activity").first
```

Comment:
As I said in my commit. I really didn't want to do this conversion each time. I'd love to give the class a `Nokogiri::XML::Element` object but I could not figure out how to create one!

Reply:
`Nokogiri::XML(activity_xml, nil, "UTF-8").xpath("/iati-activities/iati-activity")` will return an `Nokogiri::XML::Element`, and you can then pass that into LegacyActivity.

You'd have to adjust `lib/legacy_activity.rb:21` to be `identifier_element = activity_node_set.children.detect { |element| element.name.eql?("iati-identifier") }` to get this test passing.

Not sure if that's what you were after!

Reply:
That's the only way I can think to do it but it requires us to have a`activity_xml` and do all the nokogiri parsing stuff.

I'd like to avoid that and say something shorter like:

```
legacy_nodes = Nokogiri::XML::Element.new(…………)
result = service(legacy_nodes).do_thing
```

Reply:
Yeah I thought I was being too simplistic there 😁 I will have a think, but I agree maybe it's worth merging this as-is so we can move on to AMS, and circle back later to refactor this, maybe together?

Reply:
Yeah happy to come back to it if there's no obvious improvement I was missing 👍